lib/engine.js

```diff
@@ -393,52 +395,46 @@ class Engine {
-
         // create input object (type Asset)
         let transformerInput = {};
-        // asset needs to be downloaded locally
-        if(downloadAsset && input.url && !input.path) {
-
-            this[INTERNAL].timers.download.start();
-
-            // download input to <baseDirectory>/<stepNumber>-<transformerName>/inputfile and set input.path
-            transformerInput = await getSource(
-                input.url,
-                transformerDirectory.base,
-                false
-            );
-
-            this[INTERNAL].timers.download.stop();
-
-        } // url to asset is available and transformer to use it to process asset (ex: worker-ffmpeg)
-        else if(!downloadAsset && input.url) {
-            // dont download - T.compute will use url to process asset
-            // input.url is already set and is passed on to Asset object 
-            debug(`asset url available and disableInputDownload is ${disableInputDownload} : dont need to do anything`);
-            transformerInput = new Asset(input);
-        } // if (file exists locally && (T.manifest.input.sourceType === 'URL') then upload to tmp cloud
-        else if(!downloadAsset && input.path) {
-            const preSignedUrl = await this.getPresignedUrl(input.path);
-            // TODO: remove this log
-            debug('generated presigned url', preSignedUrl);
-            input.url = preSignedUrl;
-            transformerInput = new Asset(input, input.path);
-
-        } // asset is an intermediate rendition from previous transformer and is already available in input.path
-        else if(input.path) {
-            transformerInput = new Asset(input, input.path);
+        if (input && !input.url && !input.path) throw new GenericError('No source file accessible.');
+
+        debug(`sourceType: ${input.sourceType}`);
+        if (input.sourceType === TransformerSourceType.URL) {
+            const inputUrl = input.url ? input.url : '';
+            const protocol = URL.parse(inputUrl).protocol;
+            if (input.url && protocol === HTTPS_PROTOCOL) {
+                debug(`asset url available : dont need to do anything`);
+                transformerInput = new Asset(input);
+            } else if (input.url && protocol === DATA_PROTOCOL) {
+                debug(`asset url available as datauri : upload asset to a temp cloud storage and generate a presigned URL`);
+                // the will write asset locally and upload asset to a temp cloud storage to get a presigned URL
+                this[INTERNAL].timers.upload.start();
+                transformerInput = await getSource(input.url, transformerDirectory.base, true);
+                this[INTERNAL].timers.upload.stop();
+            } else {
+                debug(`asset url not available : generated presigned url`);
+                this[INTERNAL].timers.upload.start();
+                const preSignedUrl = await this.getPresignedUrl(input.path);
```

Comment:
I would put a try/catch around this, and log the errors if any happen (then throw a transformer error).

Reply:
added